docs/content/using-npm/dependency-selectors.md

```diff
@@ -58,10 +58,31 @@ The [`npm query`](/commands/npm-query) commmand exposes a new dependency selecto
 - `:extraneous` when a dependency exists but is not defined as a dependency of any node
 - `:invalid` when a dependency version is out of its ancestors specified range
 - `:missing` when a dependency is not found on disk
-- `:semver(<spec>)` matching a valid [`node-semver`](https://github.com/npm/node-semver) spec
+- `:semver([spec], [selector], [function])` match a valid [`node-semver`](https://github.com/npm/node-semver) version or range to an optional selector (defaults to `[version]`) and providing an optional semver function name
 - `:path(<path>)` [glob](https://www.npmjs.com/package/glob) matching based on dependencies path relative to the project
 - `:type(<type>)` [based on currently recognized types](https://github.com/npm/npm-package-arg#result-object)
 
+##### `:semver([spec], [selector], [function])`
+
+The `:semver()` pseudo selector allows comparing fields from each node's `package.json` using [semver](https://github.com/npm/node-semver#readme) methods. It accepts up to 3 parameters, all of which are optional.
+
+- `spec` a semver version or range (defaults to `*`)
```

Comment:
the previous default was for this pseudo selector to become a no-op if no spec is provided, see: https://github.com/npm/cli/blob/latest/workspaces/arborist/lib/query-selector-all.js#L294-L296

we are actually keeping this behavior as-is, so no breaking change. the docs are still maybe something that could be tweaked to reflect this more accurately.

Reply:
Oof - that seems like it was a bug. In-general, if a selector doesn't match it shouldn't noop & proceed to return the previous match set; it should negate that set. Selectors not delimitated by commas compound (ie. `AND` vs. `OR`). We should definitely do a separate investigation to make sure that behavior isn't any where else.

Reply:
i agree it's weird behavior, just to play things safe i'm adding a warning log that usage without any parameters is deprecated. since npm@9 is right around the corner we can break it almost right away without any surprises. i'll clean up the docs to remove mention of this usage too